src/mon/MonmapMonitor.h

```diff
@@ -29,17 +32,74 @@
 #include "MonMap.h"
 #include "MonitorDBStore.h"
 
+#include "common/command_multiplexer.h"
+
 class MMonGetMap;
 class MMonMap;
 class MMonCommand;
 class MMonJoin;
 
+namespace ceph_cmds {
+
+/* This is in effect a closure that commands operate in. The type is abitrary
+and you can do whatever you like (including the work currently done in the 
+implementation-- it could be consolidated into here): */
+struct mon_command_ctx
+{
+ struct exception : public std::runtime_error
+ {
+     const int error_code;
+
+     exception(const int error_code_, const char* what_)
+      : std::runtime_error(what_),
+        error_code(error_code_)
+     {}
+
+     exception(const int error_code_, const std::string&& what_)
+      : exception(error_code_, what_.c_str())
+     {}
+ };
+
+ public:
+ MonOpRequestRef&               op;
+ MonmapMonitor&                 monmap_monitor; 
+ Monitor&                       mon;
+ MonMap*&                       monmap;         // aka "p", mon->monmap (sometimes mutated by side-effect)
+ MonMap&                        pending_map;
+ MMonCommand&                   mmon_command;   // aka "m"
+ MonSession*                    session;        // observing ptr
+ cmdmap_t                       cmdmap;
+ std::string                    prefix;
+ std::unique_ptr<Formatter>     fptr;           // JFW: why can't we have static Formatters..?
+ Formatter*                     formatter;      // aka f; observing ptr 
+ mutable std::stringstream      ss;
```

Comment:
Part of the intent here is that we can /see/ what parameters are or are not const.